diff --git a/html/src/components/terminal/xterm/index.ts b/html/src/components/terminal/xterm/index.ts
--- a/html/src/components/terminal/xterm/index.ts
+++ b/html/src/components/terminal/xterm/index.ts
@@ -124,4 +124,6 @@ export class Xterm {
     @bind
     public focus() {
-        this.terminal?.focus();
+        if (!window.matchMedia('(hover: none), (pointer: coarse), (max-width: 768px)').matches) {
+            this.terminal?.focus();
+        }
     }
